backend/app/controllers/books_controller.rb

```diff
@@ -2,6 +2,18 @@
 
 # Books Controller
 class BooksController < ApplicationController
+  before_action :set_book, only: [:show]
```

Comment:
You can remove this if you delegate the set_book to a memoized function. Although callbacks are powerful and useful let's avoid them when we can :)